pkg/sentry/fsimpl/devpts/terminal.go

```diff
@@ -105,6 +111,8 @@ func (tm *Terminal) setForegroundProcessGroup(ctx context.Context, args arch.Sys
 		return 0, err
 	}
 
+	tm.fgProcessGroup = task.ThreadGroup().ProcessGroup()
```

Comment:
ThreadGroup.ProcessGroup() is different than the ForegroundProcessGroup.

We have the pgid for the intended new foreground process group. We can look up the actual group from that with:

`fgProcessGroup := task.PIDNamespace().ProcessGroupWithID(pgid)`

But we should only set that if the call to `task.ThreadGroup.SetForegroundProcessGroup()` succeeds:


```
ret, err := task.ThreadGroup().SetForegroundProcessGroup(tm.tty(isMaster), kernel.ProcessGroupID(pgid)
if err == nil {
  fgProcessGroup := task.PIDNamespace().ProcessGroupWithID(pgid) 
}
return ret, err
```

Reply:
👍